tests/test_migrator.py

```diff
@@ -1,9 +1,35 @@
-# -*- coding: utf-8 -*-
+# from mock import patch
 
-import unittest
+from migrate_metedata.migrator import Migrator
+import metastore.backend as store
 
-import ckan_datapackage_tools.converter as converter
+migrator_obj = Migrator('https://demo.ckan.org/api/3/action', 'filesystem', dict(uri='mem://'))
 
 
-class TestMigrateToGithub(unittest.TestCase):
-    pass
+class TestMigrator:
+
+    def test_create_url_with_pkg(self):
+        suffix = '/package_show?id='
+        pkg_name = 'world-population'
+        url = 'https://demo.ckan.org/api/3/action' + suffix + pkg_name
+
+        assert migrator_obj.create_url(suffix, pkg_name) == url
+
+    def test_create_url_without_pkg(self):
+        suffix = '/package_list'
+        url = 'https://demo.ckan.org/api/3/action' + suffix
+
+        assert migrator_obj.create_url(suffix) == url
+
+    def test_create_pkg(self):
+        data = {"result": {"name": "test_pkg",
+               "resources": [
+                   {"path": "data/myresource.csv"}
+               ]}}
+        pkg_name = "test_pkg"
+
+        assert migrator_obj.create_pkg(data, pkg_name) == True
+
+    # @patch('Migrator.urllib.urlopen')
```

Comment:
Sometimes monkey patching is a must, but typically in new code if you have to do it it is a likely code smell.

As I suggested above, if you had the ability to inject a "CKAN Client" object to the Migrator constructor, you would not need any patching here. You could just create a "mock" client (e.g. inherit your CKAN client class for testing purposes and create a "TestCKANClient" which overrides a single `_send_request()` method to return a pre-defined response instead of making an actual API call. Then pass that to the constructor instead of the real CKAN client, and boom you can test it all without any patching or mocking. 

The goal is, if you think about it, a bit like I said earlier: split away IO and flow control from the parts that you really care about (business logic). Make sure business logic is extremely easy to test. Then make it easy to mock / patch / replace the code that does IO for testing purposes. Then, if you *have* to test the IO stuff (i.e. it has code that you created and is not tested by the vendor and is not trivial), you can use dedicated tools like VCR or monkey-patching.